core/bridges/bridge_cache.go

```diff
@@ -0,0 +1,290 @@
+	return c.orm.BridgeTypes(ctx, offset, limit)
+}
+
+func (c *BridgeCache) CreateBridgeType(ctx context.Context, bt *BridgeType) error {
+	err := c.orm.CreateBridgeType(ctx, bt)
+	if err != nil {
+		return err
+	}
+
+	c.bridgeTypesCache.Store(bt.Name, *bt)
+
+	return nil
+}
+
+func (c *BridgeCache) UpdateBridgeType(ctx context.Context, bt *BridgeType, btr *BridgeTypeRequest) error {
+	if err := c.orm.UpdateBridgeType(ctx, bt, btr); err != nil {
+		return err
+	}
+
+	c.bridgeTypesCache.Store(bt.Name, *bt)
+
+	return nil
+}
+
+func (c *BridgeCache) GetCachedResponse(ctx context.Context, dotId string, specId int32, maxElapsed time.Duration) ([]byte, error) {
+	// prefer to get latest value from cache
+	c.mu.RLock()
+	cached, inCache := c.bridgeLastValueCache[responseKey(dotId, specId)]
+	c.mu.RUnlock()
+
+	if inCache && cached.FinishedAt.After(time.Now().Add(-maxElapsed)) {
+		return cached.Response, nil
+	}
+
+	response, finishedAt, err := c.orm.GetCachedResponseWithFinished(ctx, dotId, specId, maxElapsed)
+	if err != nil {
+		return nil, err
+	}
+
+	c.mu.Lock()
+	c.bridgeLastValueCache[responseKey(dotId, specId)] = BridgeResponse{
+		DotId:      dotId,
+		SpecId:     specId,
+		Response:   response,
+		FinishedAt: finishedAt,
+	}
+	c.mu.Unlock()
+
+	return response, nil
+}
+
+func (c *BridgeCache) UpsertBridgeResponse(ctx context.Context, dotId string, specId int32, response []byte) error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	c.bridgeLastValueCache[responseKey(dotId, specId)] = BridgeResponse{
+		DotId:      dotId,
+		SpecId:     specId,
+		Response:   response,
+		FinishedAt: time.Now(),
+	}
+
+	return nil
+}
+
+func (c *BridgeCache) ExternalInitiators(ctx context.Context, offset int, limit int) ([]ExternalInitiator, int, error) {
+	return c.orm.ExternalInitiators(ctx, offset, limit)
+}
+
+func (c *BridgeCache) CreateExternalInitiator(ctx context.Context, externalInitiator *ExternalInitiator) error {
+	return c.orm.CreateExternalInitiator(ctx, externalInitiator)
+}
+
+func (c *BridgeCache) DeleteExternalInitiator(ctx context.Context, name string) error {
+	return c.orm.DeleteExternalInitiator(ctx, name)
+}
+
+func (c *BridgeCache) FindExternalInitiator(ctx context.Context, eia *auth.Token) (*ExternalInitiator, error) {
+	return c.orm.FindExternalInitiator(ctx, eia)
+}
+
+func (c *BridgeCache) FindExternalInitiatorByName(ctx context.Context, iname string) (exi ExternalInitiator, err error) {
+	return c.orm.FindExternalInitiatorByName(ctx, iname)
+}
+
+func (c *BridgeCache) GetCachedResponseWithFinished(ctx context.Context, dotId string, specId int32, maxElapsed time.Duration) ([]byte, time.Time, error) {
+	return c.orm.GetCachedResponseWithFinished(ctx, dotId, specId, maxElapsed)
+}
+
+func (c *BridgeCache) BulkUpsertBridgeResponse(ctx context.Context, responses []BridgeResponse) error {
+	return c.orm.BulkUpsertBridgeResponse(ctx, responses)
+}
```

Comment:
Is it possible to embed `ORM` directly so that these are automatically available, and we just override a subset?